### PR TITLE
ZEN-32233 Information Disclosure - Server Version Headers

### PR DESCRIPTION
**Pull request by romanbakaleyko**

### CODE REVIEW

File: migrations/src/zenservicemigration/migrations/hide_nginx_version.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+                return
+
+       insert_point = insert_point.end()
+       config_text = config_text[:insert_point] + "\n" + directive + config_text[insert_point:]
+       log.info("Adding 'server_tokens off' to '%s' for '%s'", configfile.name, zproxy.name)
+       configfile.content = config_text
+       changed = True
+
+   if changed:
+        ctx.commit()
```
Comment (jpeacock-zenoss) [inline migrations/src/zenservicemigration/migrations/hide_nginx_version.py:45]: This is unnecessary.  The framework will commit for you when you return True.  So you can end the function with
```
    return changed
```

File: migrations/src/zenservicemigration/migrations/hide_nginx_version.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    for configfile in filter(lambda f: f.name == '/opt/zenoss/zproxy/conf/zproxy-nginx.conf', configfiles):
+        config_text = configfile.content
+        
+        if directive in config_text:
+            continue
+
+        insert_point = re.search('http \{', config_text)
+        if not insert_point:
+            log.info("Couldn't update zproxy configuration, skipping.")
+                return
```
Comment (jpeacock-zenoss) [inline migrations/src/zenservicemigration/migrations/hide_nginx_version.py:36]: Should not this `return` be a `continue` instead?  That would be in keeping with the use of the `for` loop to iterate over the services.

Comment (romanbakaleyko) [reply migrations/src/zenservicemigration/migrations/hide_nginx_version.py]: right, done

File: migrations/src/zenservicemigration/migrations/hide_nginx_version.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    for configfile in filter(lambda f: f.name == '/opt/zenoss/zproxy/conf/zproxy-nginx.conf', configfiles):
+        config_text = configfile.content
+        
+        if directive in config_text:
+            continue
+
+        insert_point = re.search('http \{', config_text)
+        if not insert_point:
+            log.info("Couldn't update zproxy configuration, skipping.")
+                continue
```
Comment (jpeacock-zenoss) [inline migrations/src/zenservicemigration/migrations/hide_nginx_version.py:36]: The `continue` should be lined up with the `log.info(...` statement.

**Pull request merged**